MCLauncher/MainWindow.xaml.cs

```diff
@@ -263,20 +293,28 @@ public class Versions : List<Object> {
         public class Version : NotifyPropertyChangedBase {
 
             public Version() { }
-            public Version(string uuid, string name, bool isBeta, ICommonVersionCommands commands) {
+            public Version(string uuid, string name, bool isBeta, ICommonVersionCommands commands, string directory = "") {
```

Comment:
Why "" instead of null?
I'd argue that in this case a separate constructor might be better as well.